src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -58,37 +57,36 @@ public ExpenseOD addExpenseInMonth(ExpenseID expenseIDAdd) throws EntityExceptio
             double valueDouble = toDouble(expenseIDAdd.getValue());
 
             Category selectedCategory = findCategory(categoryList, expenseID.getOldCategory());
-            Category other = findCategory(categoryList, "Other");
 
             if (valueDouble < 0) {
-                // 2. Expense budget less than 0 fail
+                // 2. Expense value less than 0 fail
                 ExpenseOD expenseODFailAdd = new ExpenseOD("Expense value can't be less than $0. Please try again!");
                 return expenseOB.fail(expenseODFailAdd);
             }
-            for (Expense expense1 : monthExpenseList) {
-                if (expense1.getName().equals(expenseID.getName())) {
-                    //Repeated name in month
-                    ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
-                    return expenseOB.fail(expenseODFailEdit);
-                }
-            }
-
             if (expenseID.getIsRecurringExpense()) {
                 if(recurringExpenseList.size() > 0) {
                     for (Expense expense1 : recurringExpenseList) {
                         if (expense1.getName().equals(expenseID.getName())) {
-                            //Repeated name in recurringData
+                            //Repeated recurring expense name fail: When a user tries to add new recurring expense but the name already exist as a recurring expense.
                             ExpenseOD expenseODFailAdd = new ExpenseOD("There is a recurring expense with this name, you don't need to add expense in month. " +
                                     "(If this is not the same expense, please use another name!)");
                             return expenseOB.fail(expenseODFailAdd);
                         }
                     }
                 }else{
-                    Expense newrecurExpense = new Expense(expenseID.getName(), other, valueDouble);
+                    Expense newrecurExpense = new Expense(expenseID.getName(), selectedCategory, valueDouble);
                     session.addRecurExpense(newrecurExpense);
+                    month.addExpense(newrecurExpense);
                     ExpenseOD expenseODSuccessAdd = new ExpenseOD("You have created a new recurring expense!");
                     return expenseOB.success(expenseODSuccessAdd);}
-                    }
+            }
+            for (Expense expense1 : monthExpenseList) {
+                if (expense1.getName().equals(expenseID.getName())) {
+                    //Repeated name in month
+                    ExpenseOD expenseODFailEdit = new ExpenseOD("There is already a expense with this new name in this month.");
+                    return expenseOB.fail(expenseODFailEdit);
+                }
+            }
 
             Expense newExpense = new Expense(expenseID.getName(), selectedCategory,valueDouble);
```

Comment:
I think I'll just have the ExpenseOD instantiation and the return outside, it shows an error for not having a return statement, and we know it's not part of the exceptions that we're dealing with!

Reply:
Sounds good!

Reply:
Nvm the last if statement has an else, so these lines will have to be independent

Reply:
I see what the confusion is here, I am suggesting to move the for loop back where it was (it needs to be there otherwise we arent checking for the same name in the recurring expenses) and then have this code go in the else block that comes after the if is recurring expense. Like how i said it in the overall comment! But i think the reason why you are saying there is no else is because the for loop is between the code (unless I am misunderstanding!).

Reply:
No yah you're right I moved the previous lines into the wrong else, now they are in the else.